notebook_data_redirector/webhook_receiver.py

```diff
@@ -45,19 +47,27 @@ def lambda_handler(event, context):
         LOGGER.critical("Received invalid webhook request")
         return STATUS_SUCCESS
 
-    if trigger in common.HANDLED_FILE_TRIGGERS:
+    if (trigger in common.HANDLED_FILE_TRIGGERS) and (box_type == "file"):
         file = common.get_file(client, box_id)
         if not file:
             LOGGER.warning("File %s is missing (trashed or deleted)", box_id)
             # We don't know what the file's path was, so we'll just have to
             # let the sync lambda clean up DynamoDB.
             return STATUS_SUCCESS
 
+        # if the file isn't public but any parent directory is, make a shared link
+        if (not common.is_box_file_public(file)) and (common.is_any_parent_public(client, file)):
```

Comment:
Since we're not caching inside of `is_any_parent_public` anymore, we probably should save the result rather than calculate it twice